### DISCUSSION TITLE
Piecewise Linear Detrending

### DISCUSSION THREAD

**Discussion by sbuse** [alan-turing-institute/sktime#5197] [General]
Dear developers, dear community

First of all I would like to thank you for providing such an amazing forecasting tool. I have been experimenting with composed forecasters and I was wondering if anyone has ever done a piecewise-linear-detrending similar to the trend fitting in the prophet model?

I know i can build a custom forecaster with regressors from sklearn but i couldn't find any piecewise linear regressor in their arsenal. I'm aware that there are several options like the polynomial detrending but i'm looking for something like this: https://github.com/chasmani/piecewise-regression

Here is an example of what i'm trying to do: 

```python
from sktime.forecasting.compose             import TransformedTargetForecaster
from sktime.forecasting.trend               import TrendForecaster
from sktime.forecasting.arima               import ARIMA
from sklearn.linear_model                   import Lasso #replace this with piecewise-linear-regressor


forecaster = TransformedTargetForecaster(
    [
        ("detrend", Detrender(forecaster=TrendForecaster(Lasso()))),
        ("forecast", ARIMA() )  
    ]
)

forecaster.fit(y)
```

**Comment by fkiraly** [comment]
@sbuse, interesting idea.

Could you explain, how would the forecast differ for a piecewise linear detrender, compared to simply fitting a linear trend to some most recent window? Would that not add back a linear trend?

**Comment by sbuse** [comment]
You are right @fkiraly about the most recent window. If one would fit a linear trend to the most recent window we would get the same linear trend but for that, we would need to know the changepoint (where to start the most recent window) and we could also not use the detrended data before the changepoint for the model building.
